cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb

```diff
@@ -89,61 +89,81 @@ def self.generate_rcinfo()
         @@configuration.select { |key, value| !CUSTOM_CONFIGURATION_KEYS.include?(key) }
       end
 
-      def self.enable_xcremotecache(target, user_proj_directory, xc_location, xc_cc_path, mode, exclude_build_configurations, check_build_configuration, check_platform, final_target)
-          target.build_configurations.each do |config| 
-            # apply only for relevant Configurations
-            next if exclude_build_configurations.include?(config.name)
-            if mode == 'consumer'
-              config.build_settings['CC'] = [xc_cc_path]
-            end
-            config.build_settings['SWIFT_EXEC'] = ["#{xc_location}/xcswiftc"]
-            config.build_settings['LIBTOOL'] = ["#{xc_location}/xclibtool"]
-            config.build_settings['LD'] = ["#{xc_location}/xcld"]
-
-            config.build_settings['XCREMOTE_CACHE_FAKE_SRCROOT'] = FAKE_SRCROOT
-            add_cflags!(config.build_settings, '-fdebug-prefix-map', "#{user_proj_directory}=$(XCREMOTE_CACHE_FAKE_SRCROOT)")
-            add_swiftflags!(config.build_settings, '-debug-prefix-map', "#{user_proj_directory}=$(XCREMOTE_CACHE_FAKE_SRCROOT)")
+      def self.enable_xcremotecache(target, xc_location, xc_cc_path, mode, exclude_build_configurations, final_target)
+        target.build_configurations.each do |config|
+          # apply only for relevant Configurations
+          next if exclude_build_configurations.include?(config.name)
+          if mode == 'consumer'
+            config.build_settings['CC'] = ["$SRCROOT/#{xc_cc_path}"]
           end
-        
+          config.build_settings['SWIFT_EXEC'] = ["$SRCROOT/#{xc_location}/xcswiftc"]
+          config.build_settings['LIBTOOL'] = ["$SRCROOT/#{xc_location}/xclibtool"]
+          config.build_settings['LD'] = ["$SRCROOT/#{xc_location}/xcld"]
 
-        # User project is not generated from scratch (contrary to `Pods`), delete all previous XCRemoteCache phases
-        target.build_phases.delete_if {|phase| 
-          # Some phases (e.g. PBXSourcesBuildPhase) don't have strict name check respond_to?
-          if phase.respond_to?(:name) 
-              phase.name != nil && phase.name.start_with?("[XCRC]")
-          end
-        }
+          config.build_settings['XCREMOTE_CACHE_FAKE_SRCROOT'] = FAKE_SRCROOT
+          add_cflags!(config.build_settings, '-fdebug-prefix-map', "$SRCROOT=$(XCREMOTE_CACHE_FAKE_SRCROOT)")
```

Comment:
We need to make sure to always have consistent prefix mapping. For `Pods` project, previously we have `ProjectRoot`->`fake` and now we would have `ProjectRoot/Pods`->`fake`. That would destroy debugging experience for Pods projects. 

To align paths, we can employ nice Xcode's BS evaluation for a dir parent. 
```suggestion
          add_cflags!(config.build_settings, '-fdebug-prefix-map', "$(SRCROOT:dir:standardizepath)=$(XCREMOTE_CACHE_FAKE_SRCROOT)")
```

So sometimes you have to replace `user_proj_directory` with `$(SRCROOT)../` (for a Pods project) and sometimes to `$(SRCROOT)` (user's targets).